src/containers/random-pick/random-pick-playground/playgrund-modal/modal-result/modal-result.tsx

```diff
@@ -0,0 +1,75 @@
+import {
+  useRandomPickPlaygroundAction,
+  useRandomPickPlaygroundState,
+} from '@/containers/random-pick/random-pick-playground-provider.tsx/random-pick-playground-provider.hooks';
+import { XIcon } from 'lucide-react';
+import { Button } from '@/components/button';
+import { MODAL_STATE_TYPES } from '@/containers/random-pick/random-pick-playground-provider.tsx/random-pick-playground-provider.constans';
+import { z } from 'zod';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Form, FormControl, FormField, FormMessage } from '@/components/form';
+import { Input } from '@/components/input';
+import ResultCard from '../../playground-card/playground-result-card';
+
+export default function ResultModal() {
+  const { winners, maxNumberOfPick } = useRandomPickPlaygroundState();
+  const { selectModalState, runPick } = useRandomPickPlaygroundAction();
+
+  const formSchema = z.object({
+    number: z.coerce
+      .number()
+      .min(1, {
+        message: '최소 인원은 1명입니다.',
+      })
+      .max(maxNumberOfPick, {
+        message: `남은 인원은 ${maxNumberOfPick}명입니다.`,
+      }),
+  });
```

Comment:
```suggestion
  const getFormSchema = (max: number) => z.object({
    number: z.coerce
      .number()
      .min(1, {
        message: '최소 인원은 1명입니다.',
      })
      .max(max, {
        message: `남은 인원은 ${max}명입니다.`,
      }),
  });
  
  const formSchema = getFormSchema(maxNumberOfPick);
```

회의 때 말씀드리려 한 게 이거였어요 ㅎㅎ 이렇게 함수로 바꾸면 `getFormSchema` 함수는 원래대로 컴포넌트 바깥에 두고, 컴포넌트 안에서 `maxNumberOfPick`만 인자로 넣어주면 사용할 수 있으니까요 :)